src/scripts/project/update-deps/dependency-worker.ts

```diff
@@ -0,0 +1,124 @@
+        dependencyTypes.forEach(dependencyType => {
+            const dependenciesMap = packageJson[dependencyType];
+
+            if (!dependenciesMap) {
+                return;
+            }
+
+            const dependenciesToUpdate = this.findDependenciesToUpdate(dependenciesMap, packages, isFullUpdate);
+            if (!dependenciesToUpdate.length) {
+                return;
+            }
+
+            const latestDependencies = this.findLatestDependencies(dependenciesMap);
+
+            // eslint-disable-next-line no-console
+            console.log('dependencies to update', dependenciesToUpdate);
+
+            const dependenciesWithVersion = this.getDependenciesWithVersion(dependenciesMap, dependenciesToUpdate);
+            this.updateDependencies(dependenciesToUpdate, dependenciesWithVersion, dependencyType);
+            this.restoreLatestDependencies(dependencyType, latestDependencies);
+        });
+    }
+
+    private findLatestDependencies(dependencyStore: Dictionary<string>): string[] {
+        return Object.keys(dependencyStore)
+            .filter(key => dependencyStore[key] === 'latest');
+    }
+
+    private findDependenciesToUpdate(
+        dependencyStore: Dictionary<string>,
+        packages: string[],
+        isFullUpdate: boolean
+    ): string[] {
+        if (!isFullUpdate) {
+            return packages.filter(dependency => dependencyStore[dependency]);
+        }
+
+        const snapshotPattern = /-/;
+        return Object.keys(dependencyStore)
+            .filter(key => {
+                const isSnapshot = snapshotPattern.test(dependencyStore[key]);
+                const isUrl = dependencyStore[key].startsWith('http');
+                const isLatest = dependencyStore[key] === 'latest';
+                return isSnapshot || isUrl || isLatest;
+            });
+    }
+
+    private getDependenciesWithVersion(dependencyStore: Dictionary<string>, dependenciesToUpdate: string[]): string[] {
+        return dependenciesToUpdate.map(packageToRemove => `${packageToRemove}@${dependencyStore[packageToRemove]}`);
+    }
+
+    private updateDependencies(
+        dependenciesToUpdate: string[],
+        dependenciesWithVersion: string[],
+        dependencyType: string): void {
+        const dependencies = dependenciesToUpdate.join(' ');
+        this.run(`yarn remove ${dependencies}`);
+
+        const depsWithVersion = dependenciesWithVersion.join(' ');
+        this.run(`yarn add ${depsWithVersion} ${DependencyWorker.DEPENDENCY_FLAGS.get(dependencyType)}`);
+    }
+
+    private restoreLatestDependencies(dependencyType: string, latestDependencies: string[]): void {
+        if (!latestDependencies.length) {
+            return;
+        }
+
+        const parsedPackageJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
+        const latestDependenciesMap = latestDependencies.reduce((acc, curr) => {
+            return { ...acc, [curr]: 'latest' };
+        }, {} as Dictionary<string>);
```

Comment:
Anton is always saying about usage reduce method to get dictionary. He would be upset :(